### PR TITLE
Add CSRF protection

### PR DESCRIPTION
**Pull request by Shane32**
Requires CORS preflight requests by ensuring that the requests are not 'simple' - e.g. GET request or form-POST requests - or that a specific header has been added to the request.

See similar logic: https://www.apollographql.com/docs/apollo-server/security/cors/#preventing-cross-site-request-forgery-csrf

The chosen default header was based on the current proposal by the GraphQL working group.  See:
- https://github.com/graphql/graphql-over-http/pull/303

### CODE REVIEW

File: tests/Transports.AspNetCore.Tests/Middleware/GetTests.cs [hunk 1]
```diff
@@ -65,6 +67,46 @@ public async Task BasicTest()
...
+    [InlineData("Header3", false)]
+    [InlineData("GraphQL-Require-Preflight", false)]
+    public async Task CsrfCustomTests(string? header, bool success)
+    {
+        _options.CsrfProtectionEnabled = true;
+        _options.CsrfProtectionHeaders = ["Header1", "Header2"];
+        var client = _server.CreateClient();
+        using var request = new HttpRequestMessage(HttpMethod.Get, "/graphql?query={count}");
+        if (header != null)
+            request.Headers.Add(header, "true");
```
Comment (gao-artur) [inline tests/Transports.AspNetCore.Tests/Middleware/GetTests.cs:102]: Please add a test with `null` and `string.Empty` values

**Review by gao-artur — APPROVED** [review]
